assets/js/tablegenerator.js

```diff
@@ -93,6 +107,119 @@ async function addClass() {
     localStorage.setItem(`class-${id}`, JSON.stringify({name:"Unnamed class", class:[]}))
 
     makeClass(id)
+    editClass(`class-${id}`)
+}
+
+function addExistingClass() {
+    if (selectedExistingClass == null) {
+        alert("Please select an existing class first, or create a new class with the button at the bottom.")
+        return
+    }
+
+    const id = selectedExistingClass["id"]
+    const name = selectedExistingClass["name"]
+    const classList = selectedExistingClass["class"]
+
+    localStorage.setItem(id, JSON.stringify({name:name, class:classList}))
+
+    makeClass(id.slice(6), name)
+    editClass(id)
+
+    selectedExistingClass = null
+}
+
+async function addClass() {
+    const existingClasses = await fetchClassList()
+
+    console.log(existingClasses)
+
+    const main = $("#table-div")[0]
+
+    main.innerHTML = ""
+
+    const topText = document.createElement("div")
+    topText.innerHTML = "Choose existing class:"
+
+    const cancelButton = document.createElement("button")
+    cancelButton.innerHTML = "CANCEL"
+    cancelButton.onclick = function() {
+        main.innerHTML = ""
+        selectedExistingClass = null
+    }
+
+    const title = document.createElement("div")
+    title.className = "class-title"
+    title.appendChild(topText)
+    title.appendChild(cancelButton)
+
+    const classes = document.createElement("div")
+    classes.id = "classes-div"
+
+    const textContainer = document.createElement("div")
+    textContainer.className = "text-area-container"
+    textContainer.appendChild(classes)
+
+    // i have given up on naming variables in an intuitive way, good luck soldiers!
+    if (existingClasses.length != 0) {
+        // bullet point list to include all existing classes
+        const existingList = document.createElement("ul")
+
+        // Add each existing class to this list
+        for (existingClass of existingClasses) {
+            const listItem = document.createElement("li")
+            listItem.innerHTML = existingClass["name"]
+
+            listItem.id = `existing-class-${existingClass["id"].slice(6)}`
+
+            // select the class on click
+            listItem.onclick = function() { setSelectedExisting(existingClass) }
+
+            existingList.appendChild(listItem)
+        }
+
+        classes.appendChild(existingList)
+    }
+
+    else {
+        classes.innerHTML = "No existing classes found, try "
+        const redirect = document.createElement("a")
+        redirect.href = `${baseurl}/sign-in`
+        redirect.innerHTML = "signing in"
+        classes.appendChild(redirect)
+        classes.innerHTML += " or creating a new class"
+    }
+
+    const createButton = document.createElement("button")
```

Comment:
Good use of document object model! No more html strings in my codebase.

Reply:
Thanks, Toby. It's clear you didn't look hard enough.